Content.Server/GameObjects/Components/Interactable/ToolComponent.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Only unused on .NET Core due to KeyValuePair.Deconstruct
+// ReSharper disable once RedundantUsingDirective
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using Content.Server.GameObjects.Components.Chemistry;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Shared.Audio;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Interactable;
+using Content.Shared.Maps;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Map;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Interactable
+{
+    [RegisterComponent]
+    public class ToolComponent : SharedToolComponent, IAfterAttack
+    {
+#pragma warning disable 649
+        [Dependency] private IEntitySystemManager _entitySystemManager;
+        [Dependency] private readonly ITileDefinitionManager _tileDefinitionManager;
+        [Dependency] private readonly IMapManager _mapManager;
+        [Dependency] private readonly IPrototypeManager _prototypeManager;
+        [Dependency] private readonly IRobustRandom _robustRandom;
+#pragma warning restore 649
+
+        private AudioSystem _audioSystem;
+        private InteractionSystem _interactionSystem;
+        private SpriteComponent _spriteComponent;
+
+        protected ToolQuality _qualities = ToolQuality.Anchoring;
+        private string _useSound;
+        private string _useSoundCollection;
+        private float _speedModifier = 1;
+
+        [ViewVariables]
+        public override ToolQuality Qualities
+        {
+            get => _qualities;
+            set
+            {
+                _qualities = value;
+                Dirty();
+            }
+        }
+
+        /// <summary>
+        ///     For tool interactions that have a delay before action this will modify the rate, time to wait is divided by this value
+        /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
+        public float SpeedModifier
+        {
+            get => _speedModifier;
+            set => _speedModifier = value;
+        }
+
+        public string UseSound
+        {
+            get => _useSound;
+            set => _useSound = value;
+        }
+
+        public string UseSoundCollection
+        {
+            get => _useSoundCollection;
+            set => _useSoundCollection = value;
+        }
```

Comment:
Couldn't you use auto-properties for those?

Reply:
Nope. Serializer can't use auto-properties for DataField()

Reply:
You actually can with the expression support that Q added now. `serializer.DataField(this, x => x.Foo, "foo", 1);`